reporter/src/main/kotlin/reporters/AttributionDocumentPdfReporter.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.reporter.reporters
+
+import java.io.File
+import java.io.OutputStream
+import java.net.URL
+import java.net.URLClassLoader
+
+import org.eclipse.sw360.antenna.attribution.document.core.AttributionDocumentGeneratorImpl
+import org.eclipse.sw360.antenna.attribution.document.core.DocumentValues
+import org.eclipse.sw360.antenna.attribution.document.core.model.LicenseInfo
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.utils.PackageConfigurationProvider
+import org.ossreviewtoolkit.model.utils.getDetectedLicensesWithCopyrights
+import org.ossreviewtoolkit.reporter.LicenseTextProvider
+import org.ossreviewtoolkit.reporter.Reporter
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.utils.AttributionDocumentPdfModel
+import org.ossreviewtoolkit.spdx.SpdxLicense
+import org.ossreviewtoolkit.utils.ProcessedDeclaredLicense
+import org.ossreviewtoolkit.utils.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.toHexString
+
+private const val DEFAULT_TEMPLATE_ID = "basic-pdf-template"
+private const val TEMPLATE_ID = "template.id"
+private const val TEMPLATE_PATH = "template.path"
+
+class AttributionDocumentPdfReporter : Reporter {
```

Comment:
I would prefer to call this `AntennaAttributionDocumentReporter`, to make clear it is using Antenna.
Please also add documentation to the class, shortly explaining what it does, similar to most other reporter classes.

Reply:
I do not like to mention the dependency in a name. It is an general attribution document generator for pdf documents. You do not mention other dependencies in names, right?

Reply:
I guess the reasoning is that there could be more attribution document generators in the future (as "attribution document" is a rather generic term), so it could make sense to somehow classify *this* type of attribution document, and that's where the mention of "Antenna" could make sense.

Reply:
I am open for suggestions, but I still do not like the name of the dependency. In Bosch terms, this document is called disclosure document, we can add a prefix like the package notice file, whatever you like.

Reply:
I might be wrong in understanding the Kotlin but is this reporter not simply a PDF version of the `NoticeByPackage` reporter?

As a user I would prefer to have one way to customize/generate my open source notices and the format Text, PDF would just be an option I pass. Either this is a reporter that follows how Antenna does things then the name `AntennaAttributionDocumentReporter` should be used. Or alternatively it follows ORT signature (@mnonnenmacher  is working on new notice generator with templates) and then it should be a format flag for `NoticeByPackage` or ``NoticeByPackage`.

Reply:
> I might be wrong in understanding the Kotlin but is this reporter not simply a PDF version of the `NoticeByPackage` reporter?

On a very basic level it is. But it offers quite advanced formatting via templates, so they are hardly comparable.

> Either this is a reporter that follows how Antenna does things then the name AntennaAttributionDocumentReporter should be used.

Ok, agreed. Let's use `AntennaAttributionDocumentReporter` to
- make clearer this is probably not the one and only reporter that will generate PDF
- better describe the type of document, i.e. Antenna-style (in that sense "Antenna" is not just the name of the dependency here, i.e. an implementation detail, but something that is visible and relevant to the user)